### PR TITLE
Create-validator

### PR DESCRIPTION
**Pull request by FarwaMuhibZada**
# Telephone-Number-Validator 🚀

## Project Description 📝

>
 This web application validates US phone numbers based on specific formats using regular expressions (regex). It provides instant feedback on whether the entered phone number is valid or not and include this elements:

- Input field (`#user-input`) to enter a phone number.
- Button (`#check-btn`) to validate the entered phone number.
- Button (`#clear-btn`) to clear the results.
- Display area (`#results-div`) to show validation results.

## Validation Rules

The app validates phone numbers based on the following formats:
- `1 555-555-5555`
- `1 (555) 555-5555`
- `5555555555`
- `555-555-5555`
- `(555)555-5555`
- `1(555)555-5555`
  
Any deviation from these formats will result in an "Invalid US number" message.
>

## Demo 📸

life demo link  https://farwamuhibzada.github.io/Telephone-Number-Validator/

Capture

### CODE REVIEW

**Comment by FarwaMuhibZada** [comment]
hello @ahzamir
I hope you are doing well  I have no partner for this week kindly request you to review my project thank you 👍🙏.

**Review by ahzamir — APPROVED** [review]
### Hi @FarwaMuhibZada 🙋‍♀️,

> Excellent work! 💯 You have done a great job implementing all the project requirements. 🥳 Your feature is complete! Congratulations!!!🎊

- **[REQUIRED]** Share your project on this [GROUP](https://www.facebook.com/groups/kick-startdev), add a simple Proud Moment text and a screenshot of your project then besides that add your project link at the end of the text as a GitHub Source code reference and post.

- Share your Project on your LinkedIn account and add a proud moment text with a screenshot.

- Well Done great design Implemented. 🎯🎉👍
image

# Status: Approved ✅
There is nothing to say other than to merge your branch. 🎉🎯✅👍😊🥳

WinkThumbsUpGIF

 ## Highlights 🌞

🎉 Great Design Implemented 
✔️ Created a professional Repo (You created a repository with your project name)
✔️ Correct use of Gitflow (You created the development branch and made it your default branch, then created and committed your work on your feature branch)
✔️ Commit messages are meaningful
✔️ Setup the linters correctly (You added the correct linters config into your project according to the language you used on your code)
✔️ No linter errors (All linters checks have been passed 🟢)
✔️ You have followed the Telephone Number Validator project task and committed using the command line
✔️ Created the pull request (From your feature branch to the development branch you created a pull request and submitted the pull request link for review)
✔️ Good PR description (You added a meaningful description to your PR about your work and the features you added on this pull request)
✔️Your code is clean
✔️Your repo has a professional README.md file
✔️ Your partner has reviewed your repo
✔️Your design is responsive

**Cheers 🥂 and Happy coding!!! 👯 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me @ahzamir in your question so I can receive the notification.**

File: script.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  } else if (regex.test(Input.value)) {
+    results.innerText = `Valid US number: ${Input.value}`;
+  } else {
+    results.innerText = `Invalid US number: ${Input.value}`;
+  }
+});
+clear.addEventListener('click', () => {
+  results.textContent = '';
+  Input.value = '';
+});
```
Comment (ahzamir) [inline script.js:18]: - [OPTIONAL] - Try to add your Event Listeners in while DOM content is loading, it helps to add all of your event listeners to be ready to the action right when your page loads and prevents any side effects below is an example to help you do that:

```javascript
document.addEventListener('DOMContentLoaded', () => {
  //In here add all of your addEventListener functions
  // don't forget to add the constants that you need in your event listeners here as the scope of this one is higher than other codes
  const Input = document.getElementById('user-input');
  const check = document.getElementById('check-btn');
  const clear = document.getElementById('clear-btn');
  const results = document.getElementById('results-div');
  check.addEventListener('click', () => {
    const regex = /^(1\s?)?(\(\d{3}\)|\d{3})([\s-]?)\d{3}([\s-]?)\d{4}$/;
    if (!Input.value) {
      alert('Please provide a phone number');
    } else if (regex.test(Input.value)) {
      results.innerText = `Valid US number: ${Input.value}`;
    } else {
      results.innerText = `Invalid US number: ${Input.value}`;
    }
  });
  clear.addEventListener('click', () => {
    results.textContent = '';
    Input.value = '';
  });
});
```

**Review by ahzamir — COMMENTED** [review]
- [OPTIONAL] - Check these optional changes implement them and then merge your branch.

File: script.js (hunk 1 above)
Comment (FarwaMuhibZada) [reply script.js]: @ahzamir thanks for your reviewing . I added an Event Listeners in while DOM content is loading. in my project

Comment (ahzamir) [reply script.js]: @FarwaMuhibZada 

- Well done nothing to say other than you can merge your branch. 🎉👍✅

**Pull request merged**